addons/web/models/base_document_layout.py

```diff
@@ -40,12 +40,20 @@ def _default_report_footer(self):
     def _default_company_details(self):
         company = self.env.company
         address_format, company_data = company.partner_id._prepare_display_address()
+        address_format = self._clean_address_format(address_format, company_data)
         # company_name may *still* be missing from prepared address in case commercial_company_name is falsy
         if 'company_name' not in address_format:
             address_format = '%(company_name)s\n' + address_format
             company_data['company_name'] = company_data['company_name'] or company.name
         return Markup(nl2br(address_format)) % company_data
 
+    def _clean_address_format(self, address_format, company_data):
+        missing_company_data = [k for k, v in company_data.items() if not v]
+        for key in missing_company_data:
+            if key in address_format:
+                address_format = ''.join(address_format.split(f'%({key})s\n'))
```

Comment:
Isn't this split + join a complicated way to remove some substrings from the string?

```suggestion
                address_format = address_format.replace(f'%({key})s\n', '')
```

Are we sure of the format? It is really always `%s(...)s\n`? (real question, I'm not a PDF address expert)

Reply:
I would say I'm "pretty sure" :wink: 
It comes from here : 
https://github.com/odoo/odoo/blob/15.0/odoo/addons/base/models/res_partner.py#L983-L985

Reply:
https://github.com/odoo/odoo/blob/06c8765d362f81aa92f25029cb5d4e9d4069b3d3/odoo/addons/base/models/res_partner.py#L983-L985

So it is overridable and doesn't always ends with a newline (city, state_code and country_name). I don't have a good feeling about that search and replace...

Reply:
I must admit I took my inspiration from what was already being done in `_default_company_details()` :
https://github.com/odoo/odoo/blob/77fce500a9742dc2bbec6b46193e3ef1546bfe85/addons/web/models/base_document_layout.py#L44-L46
We are already relying on the same `%s(...)s\n` pattern.
So I figured it was OK to follow in those footsteps :man_shrugging: 
And with the newly added test, we should know if/when `_clean_address_format()` stops working.
But maybe you have another idea/suggestion ?